client/components/domains/register-domain-step/index.jsx

```diff
@@ -274,7 +274,15 @@ const RegisterDomainStep = React.createClass( {
 		async.parallel(
 			[
 				callback => {
-					if ( ! domain.match( /.{3,}\..{2,}/ ) ) {
+					if ( !domain.match( /.{3,}\..{2,}/ ) ) {
+						return callback();
+					}
+
+					/**
```

Comment:
Not sure this comment is really needed - I think the condition `this.props.isSignupStep && domain.match( /\.wordpress\.com$/ )` is pretty much self-explanatory.